edx_django_utils/monitoring/code_owner/utils.py

```diff
@@ -0,0 +1,114 @@
+
+    module_parts = module.split('.')
+    # To make the most specific match, start with the max number of parts
+    for number_of_parts in range(len(module_parts), 0, -1):
+        partial_path = '.'.join(module_parts[0:number_of_parts])
+        if partial_path in _PATH_TO_CODE_OWNER_MAPPINGS:
+            code_owner = _PATH_TO_CODE_OWNER_MAPPINGS[partial_path]
+            return code_owner
+    return None
+
+
+def is_code_owner_mappings_configured():
+    """
+    Returs True if code owner mappings were configured, and False otherwise.
+    """
+    return bool(_PATH_TO_CODE_OWNER_MAPPINGS)
+
+
+def _process_code_owner_mappings():
+    """
+    Processes the CODE_OWNER_MAPPINGS Django Setting and returns a dict optimized
+    for efficient lookup by path.
+
+    Returns:
+         (dict): optimized dict for success processing, None if there are no
+            configured mappings, or _INVALID_CODE_OWNER_MAPPING if there is an
+            error processing the setting.
+
+    Example CODE_OWNER_MAPPINGS Django Setting::
+
+        CODE_OWNER_MAPPINGS = {
+            'team-red': [
+                'xblock_django',
+                'openedx.core.djangoapps.xblock',
+            ],
+            'team-blue': [
+                'badges',
+            ],
+        }
+
+    Example return value::
+
+        {
+            'xblock_django': 'team-red',
+            'openedx.core.djangoapps.xblock': 'team-red',
+            'badges': 'team-blue',
+        }
+
+    """
+    _CODE_OWNER_MAPPINGS = getattr(settings, 'CODE_OWNER_MAPPINGS', None)
+    if not _CODE_OWNER_MAPPINGS:
+        return None
+
+    try:
+        path_to_code_owner_mappings = {}
+        for code_owner in _CODE_OWNER_MAPPINGS:
+            path_list = _CODE_OWNER_MAPPINGS[code_owner]
+            for path in path_list:
+                path_to_code_owner_mappings[path] = code_owner
+                optional_module_prefix_match = _OPTIONAL_MODULE_PREFIX_PATTERN.match(path)
+                # if path has an optional prefix, also add the module name without the prefix
+                if optional_module_prefix_match:
+                    path_without_prefix = path[optional_module_prefix_match.end():]
+                    path_to_code_owner_mappings[path_without_prefix] = code_owner
+
+        return path_to_code_owner_mappings
+    except Exception as e:
+        log.exception('Error processing code_owner_mappings. {}'.format(e))
+        # errors should be unlikely due do scripting the setting values.
+        # this will trigger an error custom metric that can be alerted on.
+        return _INVALID_CODE_OWNER_MAPPING
+
+# .. toggle_name: CODE_OWNER_MAPPINGS
+# .. toggle_implementation: DjangoSetting
+# .. toggle_default: None
+# .. toggle_description: Used to set monitoring custom metrics for owner. Dict with keys of code owner and value as list of dotted path module names owned by code owner.
```

Comment:
I'm getting a `line-too-long` pylint error, but doesn't this[ ignore-long-lines rule](https://github.com/edx/edx-django-utils/blob/13faff8680eebd21bab36d5d9b3a0c77c17283e1/pylintrc#L370) in pylintrc skip comments? Is it somehow not using `pylintrc`?

Reply:
Ned answered my question.  There is no OR in the rule, even though I wanted one. :)